src/otx/algo/classification/torchvision_model.py

```diff
@@ -23,422 +30,128 @@
-
-        saliency_map = self.explainer.func(backbone_feat)
-        feature_vector = feature_vector_fn(backbone_feat)
-
-        x = self._flatten_outputs(self.avgpool(backbone_feat))
-        if self.neck is not None:
-            x = self.neck(x)
-        logits = self.head(x)
-        pred_results = self.head.predict(x) if hasattr(self.head, "predict") else self.softmax(logits)
-
-        outputs = {
-            "logits": logits,
-            "feature_vector": feature_vector,
-            "saliency_map": saliency_map,
-        }
-
-        if not torch.jit.is_tracing():
-            if isinstance(pred_results, dict):
-                outputs["scores"] = pred_results["scores"]
-                outputs["preds"] = pred_results["labels"]
-            else:
-                outputs["scores"] = pred_results.unbind(0)
-                outputs["preds"] = logits.argmax(-1, keepdim=True).unbind(0)
-
-        return outputs
-
-    @torch.no_grad()
-    def _head_forward_fn(self, x: torch.Tensor) -> torch.Tensor:
-        """Performs model's neck and head forward."""
-        x = self._flatten_outputs(self.avgpool(x))
-        if self.neck is not None:
-            x = self.neck(x)
-        return self.head(x)
-
-    def _flatten_outputs(self, x: torch.Tensor) -> torch.Tensor:
-        # Flatten the output
-        if len(x.shape) == 4 and not self.use_layer_norm_2d:  # If feats is a 4D tensor: (b, c, h, w)
-            x = x.view(x.size(0), -1)  # Flatten the output of the backbone: (b, f)
-        return x
-
-
-class OTXTVModel(OTXModel):
-    """OTXTVModel is that represents a TorchVision model for classification.
-
-    Args:
-        backbone (TVModelType): The backbone architecture of the model.
-        label_info (LabelInfoTypes): The number of classes for classification.
-        optimizer (OptimizerCallable, optional): The optimizer to use for training.
-            Defaults to DefaultOptimizerCallable.
-        scheduler (LRSchedulerCallable | LRSchedulerListCallable, optional): The learning rate scheduler to use.
-            Defaults to DefaultSchedulerCallable.
-        metric (MetricCallable, optional): The metric to use for evaluation. Defaults to MultiClassClsMetricCallable.
-        torch_compile (bool, optional): Whether to compile the model using TorchScript. Defaults to False.
-        freeze_backbone (bool, optional): Whether to freeze the backbone model. Defaults to False.
-        task (Literal[OTXTaskType.MULTI_CLASS_CLS, OTXTaskType.MULTI_LABEL_CLS, OTXTaskType.H_LABEL_CLS], optional):
-            The type of classification task.
-        train_type (Literal[OTXTrainType.SUPERVISED, OTXTrainType.SEMI_SUPERVISED], optional): The type of training.
-        input_size (tuple[int, int], optional):
-            Model input size in the order of height and width. Defaults to (224, 224)
-    """
-
-    model: TVClassificationModel
+class TVModelForMultilabelCls(OTXMultilabelClsModel):
```

Comment:
Could you add argument docstring?

Reply:
Thanks! Done - https://github.com/openvinotoolkit/training_extensions/commit/5ed3863c2f7d3db9b94acbbf9530abbce28536ba